src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/ListPackageCommandRunner.cs

```diff
@@ -0,0 +1,421 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Build.Evaluation;
+using NuGet.CommandLine.XPlat.Utility;
+using NuGet.Common;
+using NuGet.Configuration;
+using NuGet.Frameworks;
+using NuGet.ProjectModel;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+using NuGet.Versioning;
+
+namespace NuGet.CommandLine.XPlat
+{
+    public class ListPackageCommandRunner : IListPackageCommandRunner
+    {
+        private const string ProjectAssetsFile = "ProjectAssetsFile";
+        private const string ProjectName = "MSBuildProjectName";
+        private const string LeftPadding = "   ";
+
+        public async Task ExecuteCommandAsync(ListPackageArgs listPackageArgs)
+        {
+            //If the given file is a solution, get the list of projects
+            //If not, then it's a project, which is put in a list
+            var projectsPaths = Path.GetExtension(listPackageArgs.Path).Equals(".sln")?
+                           MSBuildAPIUtility.GetProjectsFromSolution(listPackageArgs.Path).Where(f => File.Exists(f)):
+                           new List<string>(new string[] { listPackageArgs.Path });
+
+            var autoReferenceFound = false;
+
+            var msBuild = new MSBuildAPIUtility(listPackageArgs.Logger);
+
+            //Loop through all of the project paths
+            foreach (var projectPath in projectsPaths)
+            {
+                //Open project to evaluate properties for the assets
+                //file and the name of the project
+                var project = MSBuildAPIUtility.GetProject(projectPath);
+
+                if (!MSBuildAPIUtility.IsPackageReferenceProject(project))
+                {
+                    Console.Error.WriteLine(string.Format(CultureInfo.CurrentCulture,
+                        Strings.Error_NotPRProject,
+                        projectPath));
+                    Console.WriteLine();
+                    continue;
+                }
+
+                var projectName = project.GetPropertyValue(ProjectName);
+
+                var assetsPath = project.GetPropertyValue(ProjectAssetsFile);
+
+                //If the file was not found, print an error message and continue to next project
+                if (!File.Exists(assetsPath))
+                {
+                    Console.Error.WriteLine(string.Format(CultureInfo.CurrentCulture,
+                        Strings.Error_AssetsFileNotFound,
+                        projectPath));
+                    Console.WriteLine();
+                    continue;
```

Comment:
Same as before, I'd say continue in the middle of a loop is a trap. 
At the beginning it's better.